lib/src/Base/IterativeStat/IterativeMoments.cxx

```diff
@@ -0,0 +1,286 @@
+void IterativeMoments::increment(const Point & newData)
+{
+  /* centered new data */
+  Point delta(newData);
+  if (iteration_ > 0)
+  {
+    /* mean at the previous step */
+    const Point mu_previous(getMean());
+    delta -= mu_previous;
+  }
+  ++ iteration_;
+  const Point delta_over_n = delta / iteration_;
+
+  for(UnsignedInteger d = 0; d < dimension_; ++d)
+  {
+    centeredMoments_(0, d) += newData[d];
+  }
+  if(orderMax_ > 1) updateHigherMoments(orderMax_, delta, delta_over_n);
+}
+
+/*  Increment method from a Sample */
+void IterativeMoments::increment(const Sample & newData)
+{
+  for(UnsignedInteger j = 0; j < newData.getSize(); ++j)
+  {
+    increment(newData[j]);
+  }
+}
+
+/* Update the centered moments higher than 2 using a recurrence relation for 3 and more
+    see: Simpler Online Updates for Arbitrary-Order Central Moments, Meng (2015)
+*/
+void IterativeMoments::updateHigherMoments(UnsignedInteger orderMax, const Point & delta, const Point & delta_over_n)
+{
+  /* special case: second centered moment */
+  if(orderMax == 2)
+  {
+    for(UnsignedInteger d = 0; d < dimension_; ++d)
+    {
+      centeredMoments_(1, d) += delta[d] * (delta[d] - delta_over_n[d]);
+    }
+  }
+  else
+  {
+    /* general case for order >= 2 */
+    for(UnsignedInteger d = 0; d < dimension_; ++d)
+    {
+      centeredMoments_(1, d) += delta[d] * (delta[d] - delta_over_n[d]);
+      for(UnsignedInteger order = 3 ; order <= orderMax ; ++order)
+      {
+        Scalar tmp = 0.0;
+        for(UnsignedInteger l = 1; l <= order - 2; ++l)
+        {
+          const UnsignedInteger binomialCoeff = SpecFunc::BinomialCoefficient(order, l) ;
+          tmp += std::pow(delta_over_n[d], l) * centeredMoments_(order - l - 1, d) * binomialCoeff ;
+        }
+        centeredMoments_(order - 1, d) += delta[d] * (std::pow(delta[d], order - 1) - std::pow(delta_over_n[d], order - 1)) - tmp;
+      }
+    }
+  }
+}
+
+/* Accessor to the coefficient of variation */
+Point IterativeMoments::getCoefficientOfVariation() const
+{
+  if ((orderMax_ < 2)) throw InternalException(HERE) << "Error: unavailable method, the declared maximum order is lower than 2.";
+
+  Point coefficientOfVariationData(dimension_, 0.0);
+  const Point meanData(getMean());
+  const Point varData(getVariance());
+  for (UnsignedInteger i = 0; i < dimension_; ++i)
+  {
+    coefficientOfVariationData[i] = std::pow(varData[i], 0.5) / meanData[i];
+  }
+
+  return coefficientOfVariationData;
+}
+
+/* Accessor to the standard error of the mean */
+Point IterativeMoments::getStandardErrorOfTheMean() const
+{
+  if ((orderMax_ < 2)) throw InternalException(HERE) << "Error: unavailable method, the declared maximum order is lower than 2.";
+
+  Point standardErrorOfTheMean(dimension_, 0.0);
+  const Point varData(getVariance());
+
+  for (UnsignedInteger i = 0; i < dimension_; ++i)
+  {
+    standardErrorOfTheMean[i] = std::pow(varData[i], 0.5) / std::pow(iteration_, 0.5);
```

Comment:
std::sqrt, better than std::pow, and take a unique square root of a ratio instead of the ratio of two square roots

Reply:
Ok. Fixed.